### PR TITLE
Add config option to disable new account creation prompt

### LINKED ISSUE #99: [Feature request] Disable new user signups
When a new user types their email address  in the signin form, it shows the new signup form.
How can I disable this? 
I just want the users who I have already created from firebase console to access my app.

### PR DESCRIPTION
**Pull request by chucklay**
Potential way to fix #99 - Adds allowNewAccountCreation config param (Which defaults to true, keeping current behavior). If set to false this displays the full username and password prompt, and does not show the "sign up" screen if an unrecognized email address is entered.

As people in the feature request thread have mentioned, new account creation should also be disabled server-side, but this prevents a lot of confusion on the part of the end user.

(Re-opening with fixed git email for CLA)

### CODE REVIEW

**Comment by chucklay** [comment]
@googlebot I signed it!

File: javascript/widgets/config.js [hunk 1]
```diff
@@ -46,6 +46,11 @@ class Config {
     this.config_.define(
         'credentialHelper',
         Config.CredentialHelper.ACCOUNT_CHOOSER_COM);
+    /**
+     * Defines if unrecognized email addresses should be prompted to create
+     * an account. If set to false, the full email/password prompt will be shown.
```
Comment (shaibruhis) [inline javascript/widgets/config.js:51]: do you mean to say if "if set to true, ..."?

File: javascript/widgets/config.js [hunk 2]
```diff
@@ -800,6 +805,14 @@ class Config {
         Config.CredentialHelper.ACCOUNT_CHOOSER_COM;
   }
 
+  /**
+   * @return {boolean} Whether new accounts should should be allowed to register through the ui
```
Comment (shaibruhis) [inline javascript/widgets/config.js:809]: you wrote the word "should" twice :)

File: javascript/widgets/config.js [hunk 3]
```diff
@@ -800,6 +805,14 @@ class Config {
...
+   */
+  isNewAccountCreationAllowed() {
+    return /** @type {boolean} If new account creation is allwed */ (
```
Comment (shaibruhis) [inline javascript/widgets/config.js:812]: spelled "allowed" wrong

**Comment by chucklay** [comment]
@shaibruhis Thanks for the feedback! Comments are harder than code sometimes. Yeah, as far as I know I'm just waiting on approval from Google! Fingers crossed!

**Comment by shaibruhis** [comment]
It looks like I can approve the PR actually. Do you have any tests or screen recordings?

**Comment by robraux** [comment]
@chkuang-g @shaibruhis any movement on getting this merged?

**Comment by AxelMooreDev** [comment]
I really need this feature. please merge.

**Comment by chucklay** [comment]
@googlebot I fixed it.

**Comment by chucklay** [comment]
@googlebot I fixed it.

**Comment by chucklay** [comment]
Whew, fixing CLAs are a pain. @robraux @AxelMooreDev Unfortunately I have no control over getting this merged in. Someone from Google needs to review/merge this.

**Comment by chucklay** [comment]
FYI for those wanting to use this now: It's possible to `npm install` (note that npm install will fail on tests if you're using python 3, but even then the actual install process will work) and `npm run build` this fork (or your own version) and load the resulting `firebaseui.js` in the dist folder in place of the one from google (you can also copy that file to your project if you don't want to include the whole fork).

**Comment by AxelMooreDev** [comment]
This line is needed for TypeScript

---
 types/index.d.ts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/types/index.d.ts b/types/index.d.ts
index 4ec7230..692faf5 100644
--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -110,6 +110,7 @@ declare namespace firebaseui.auth {
     autoUpgradeAnonymousUsers?: boolean;
     callbacks?: Callbacks;
     credentialHelper?: CredentialHelperType;
+    allowNewAccountCreation?: boolean;
     immediateFederatedRedirect?: boolean;
     popupMode?: boolean;
     queryParameterForSignInSuccessUrl?: string;

**Comment by damienkilgannon** [comment]
Would really love to have these feature merged!

**Comment by estudiocactus** [comment]
When this features will be merged?

**Comment by estudiocactus** [comment]
By the moment, here's my MEGA dirty workaround:

1- Create a function all the time listening when the firebase sign up form appears
2- Remove it and then create a div like "Don't have an account? Create one"
3- Redirect a page changing all the sign-in labels to sing-up

```
var checkExist = setInterval(function () {
  var signUpForm = document.querySelector('.firebaseui-id-page-password-sign-up')
  if (signUpForm) {
    signUpForm.style.display = 'none'
    document.querySelector('.signup-container').style.display = 'initial'
    clearInterval(checkExist);
  }
}, 1);
```

**Comment by robraux** [comment]
@estudiocactus it looks like only @shaibruhis is configured to review/merge this PR. That looks like the final blocker to get this merged. Is there anything you can share with us here @shaibruhis? Thank you!

**Comment by chucklay** [comment]
@estudiocactus @robraux Unfortunately someone from Google has to merge this - others (myself included) aren't authorized. I don't think @shaibruhis is authorized either (correct me if I'm wrong), they were just able to review the PR.

As far as I know, I've done everything I need to in order to get this merged in, but I haven't heard anything from Google. But like I said earlier, you should be able to clone and build this fork, and use the `firebaseui.js` it spits out (in the dist folder) in place of Google's.

**Comment by AntohiRobert** [comment]
Hi! I've just started using Firebase, seems like an incredible tool. It would be awesome to disable new user signups from the firebaseui-web. From what I understood while reading this issue, the most secure way right now is to build our own auth/signup process/ui instead, and the next most-secure-but-not-bulletproof would be either auto-delete new users on SignUp with cloud functions or hidden property in user claims "isAllowed". Would be nice if you merged it. Thanks!

**Comment by chucklay** [comment]
@robraux @AxelMooreDev @damienkilgannon @AntohiRobert Hey everyone, in case you didn't see, Google added their own implementation of this in their most recent release (see this comment: https://github.com/firebase/firebaseui-web/issues/99#issuecomment-794537000). As this works for our purposes, I'll be closing this PR in the next few days, and unless anyone objects, we'll probably get rid of this fork as well. Thanks for joining me on this wild ride!

**Pull request closed without merging**